src/sinks/sentry/metrics/sink.rs

```diff
@@ -8,34 +9,46 @@ use crate::{
 };
 
 pub struct SentryMetricsSink {
-    pub dsn: Option<String>,
+    pub dsn: String,
 }
 
 #[async_trait]
 impl StreamSink<Event> for SentryMetricsSink {
-    async fn run(mut self: Box<Self>, mut input: BoxStream<'_, Event>) -> Result<(), ()> {
-        let dsn = self.dsn.as_deref().unwrap_or("<missing>");
-        let _guard = sentry::init((dsn, sentry::ClientOptions {
-          release: sentry::release_name!(),
-          ..Default::default()
-        }));
+    async fn run(mut self: Box<Self>, input: BoxStream<'_, Event>) -> Result<(), ()> {
+        let mut input = input
+            // filter out any non-metric events
+            .filter_map(|event| ready(event.try_into_metric()));
 
-        while let Some(event) = input.next().await {
-            let metric = event.as_metric();
+        // https://github.com/getsentry/sentry-rust/issues/637
+        let _guard = sentry::init(self.dsn);
+
+        while let Some(metric) = input.next().await {
             let name = metric.series().name().name.clone();
             match metric.data().value() {
-                MetricValue::Counter { .. } => {
-                    Metric::count(name).send();
-                },
+                MetricValue::Counter { value } => {
+                    Metric::incr(name.clone(), *value).send();
```

Comment:
Is cloning the name really necessary since it was cloned just above?

Reply:
it's not. it is needed for the places where we have for-loops